virtcontainers/kata_agent.go

```diff
@@ -667,28 +689,26 @@ func (k *kataAgent) createContainer(pod *Pod, c *Container) (*Process, error) {
 		// (kataGuestSharedDir) is already mounted in the
 		// guest. We only need to mount the rootfs from
 		// the host and it will show up in the guest.
-		if err := bindMountContainerRootfs(kataHostSharedDir, pod.id, c.id, c.rootFs, false); err != nil {
-			bindUnmountAllRootfs(kataHostSharedDir, *pod)
-			return nil, err
+		if err = bindMountContainerRootfs(kataHostSharedDir, pod.id, c.id, c.rootFs, false); err != nil {
+			return
```

Comment:
Slightly numpty question - this func has a mixture of `return` and `return nil, err` - that doesn't read well to me - is that something that should really be consistent and cleaned up?